### PR TITLE
Added logging for Go Update check, changed notification message - #1614

### PR DESCRIPTION
**Pull request by maheshp**

### CODE REVIEW

**Comment by arvindsv** [comment]
This looks good to me.

File: server/webapp/WEB-INF/rails.new/app/controllers/api_v1/version_infos_controller.rb [hunk 1]
```diff
@@ -20,8 +20,10 @@ class VersionInfosController < ApiV1::BaseController
 
     def update_server
       go_latest_version = ApiV1::GoLatestVersion.new(params, system_environment)
-
-      return render_message(message_tampered_error_message, 400, {}) unless go_latest_version.valid?
+      unless go_latest_version.valid?
+        Rails.logger.error('[GO Update Check] Latest version update failed, version information from update server tampered.')
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/controllers/api_v1/version_infos_controller.rb]: Just `Go` not `GO`. Also `GoCD`,  not `GOCD`

File: server/src/com/thoughtworks/go/server/service/ServerVersionInfoBuilder.java [hunk 2]
```diff
@@ -21,7 +21,8 @@
 import com.thoughtworks.go.domain.exception.VersionFormatException;
 import com.thoughtworks.go.server.dao.VersionInfoDao;
 import com.thoughtworks.go.server.util.ServerVersion;
-import org.apache.log4j.Logger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (ketan) [inline server/src/com/thoughtworks/go/server/service/ServerVersionInfoBuilder.java]: And thank you for using `slf4j` :)

Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/ServerVersionInfoBuilder.java]: thanks to @jyotisingh for slf4j

**Pull request merged**